tensorflow/c/eager/c_api.cc

```diff
@@ -151,9 +151,10 @@ TF_DeviceList* TFE_ContextListDevices(TFE_Context* ctx, TF_Status* status) {
 }
 
 TFE_TensorHandle* TFE_NewTensorHandle(TF_Tensor* t) {
-  return new TFE_TensorHandle(
-      tensorflow::TensorCApi::MakeTensor(t->dtype, t->shape, t->buffer),
-      nullptr);
+  tensorflow::Tensor tensor;
+  // TODO: Add status argument and check on it.
+  tensorflow::TF_TensorToTensor(t, &tensor);
```

Comment:
I'm worried about ignoring the status result.
At the very least, we should return `nullptr` if `!status.ok()`